app/models/user.rb

```diff
@@ -103,8 +103,35 @@ def decline_invite
     update(invited_by_id: nil)
   end
 
-  private
-    def email_without_domain
-      email.split('@').first
+  def process_unit(unit_params, user_is_guest)
+    if user_is_guest # if user is guest
+      if unit = self.unit # and has a unit
```

Comment:
However I think you should still be able to refactor to one line `user_is_gues && unit = self.unit`